### DISCUSSION TITLE
Multi Language support & Ndot Issue

### DISCUSSION THREAD

**Discussion by reindex-ot** [MissingCore/Music#57] [General]
I would like to have multi language support.
If you have Crowdin, we can respond quickly.

...Ndot may not be suitable for Japanese environments. It is better to consider where you use it. Are there any problems with the font license?

**Comment by cyanChill** [comment]
I plan on adding internationalization/i18n sometimes in the future (just want to make sure there's no issues with `v1.0.0-rc.11` and then get `v1.0.0` on the Play Store). I need to do a bit of research on implementing i18n, probably simplify some text/phrases, and fix some code where we handle text overflow via ellipsis.
- I think most of the settings page will probably be left untranslated as there's a bunch of text. In addition, there's some places where it might not be needed (ie: pages where we display the license of third-party software).

I currently use Ndot "sparingly"; it's mainly used in headings (we also use it on the home page for displaying the number of tracks in the favorites playlist & subheading of the modal if there is one). I know with languages Ndot doesn't support, it'll fallback to the system font(?), so the text should still be visible. For all other text, we use [`Geist`](https://github.com/vercel/geist-font) (specifically the mono variant) since I think Nothing uses `LL Lettera Mono` in their apps and that's paid.

**Comment by The-Ladle** [comment]
I would like to add my interest in translating the app to German once the opportunity arrives

**Comment by cyanChill** [comment]
I've recently added the framework to support translations with #88. You can look at the [`Translations` doc](https://github.com/MissingCore/Music/blob/dev/docs/translations.md) for more information on how to contribute.

For now, there's no Crowdin project. If it might be easier, I'll go and make one (the GitHub PR method is kind of better as it makes crediting the translator easier).
